src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getPosts } from '../../api/posts';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export interface PostsSlice {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsSlice = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchPosts = createAsyncThunk<Post[], number | undefined>(
+  'posts/fetchPosts',
+  async (userId?: number) => {
+    if (userId) {
+      return getUserPosts(userId);
+    }
+
+    return getPosts();
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearError: state => {
+      state.hasError = false;
+    },
+    clearPosts: state => {
+      state.items = [];
+      state.loaded = false;
+      state.hasError = false;
+    },
+  },
+  extraReducers: builder => {
+    builder
+      .addCase(fetchPosts.pending, state => {
+        state.loaded = false;
+        state.hasError = false;
+      })
+      .addCase(fetchPosts.fulfilled, (state, action) => {
+        state.loaded = true;
+        state.items = action.payload;
+      })
+      .addCase(fetchPosts.rejected, state => {
+        state.hasError = true;
```

Comment:
Consider adding more detailed error handling or logging to provide better insights into what went wrong during the fetch operation.